addons/escoria-core/game/esc_logger.gd

```diff
@@ -0,0 +1,193 @@
+class ESCLoggerBase:
+	# Perform emergency savegame 
+	signal perform_emergency_savegame
+	
+	# Log file format
+	const  LOG_FILE_FORMAT: String = "log_%s_%s.log"
+	
+	# Valid log levels
+	enum { LOG_ERROR, LOG_WARNING, LOG_INFO, LOG_DEBUG, LOG_TRACE }
+
+	# A map of log level names to log level ints
+	var _level_map: Dictionary = {
+		"ERROR": LOG_ERROR,
+		"WARNING": LOG_WARNING,
+		"INFO": LOG_INFO,
+		"DEBUG": LOG_DEBUG,
+		"TRACE": LOG_TRACE,
+	}
+	
+	# Configured log level
+	var _log_level: int
+
+
+	# Constructor
+	func _init():
+		_log_level = _level_map[ESCProjectSettingsManager.get_setting(
+			ESCProjectSettingsManager.LOG_LEVEL
+		).to_upper()]
+
+
+	func formatted_message(context: String, msg: String, letter: String) -> String:
+		return "ESC ({0}) {1} {2}: {3}".format([_formatted_date(), letter, context, msg])
+
+
+	func trace(owner: Object, msg: String):
+		var context = owner.get_script().resource_path.get_file()
+		print(formatted_message(context, msg, "T"))
+
+
+	# Debug log
+	func debug(owner: Object, msg: String):
+		var context = owner.get_script().resource_path.get_file()
+		print(formatted_message(context, msg, "D"))
+
+
+	func info(owner: Object, msg: String):
+		var context = owner.get_script().resource_path.get_file()
+		print(formatted_message(context, msg, "I"))
+
+
+	# Warning log
+	func warn(owner: Object, msg: String):
+		var context = owner.get_script().resource_path.get_file()
+		push_warning(formatted_message(context, msg, "W"))
+		if ESCProjectSettingsManager.get_setting(
+			ESCProjectSettingsManager.TERMINATE_ON_WARNINGS
+		):
+			assert(false)
+			escoria.get_tree().quit()
+			
+	
+	# Error log
+	func error(owner: Object, msg: String):
+		var context = owner.get_script().resource_path.get_file()
+		push_error(formatted_message(context, msg, "E"))
+		if ESCProjectSettingsManager.get_setting(
+			ESCProjectSettingsManager.TERMINATE_ON_ERRORS
+		):
+			assert(false)
+			escoria.get_tree().quit()
```

Comment:
Suggest we add these lines as per below
print(formatted_message(context, msg, "W"))
print(formatted_message(context, msg, "E"))
as currently warnings and errors don't print in the debugger output, only the Godot console.

```suggestion
	# Warning log
	func warn(owner: Object, msg: String):
		var context = owner.get_script().resource_path.get_file()
		print(formatted_message(context, msg, "W"))
		push_warning(formatted_message(context, msg, "W"))
		if ESCProjectSettingsManager.get_setting(
			ESCProjectSettingsManager.TERMINATE_ON_WARNINGS
		):
			assert(false)
			escoria.get_tree().quit()
			
	
	# Error log
	func error(owner: Object, msg: String):
		var context = owner.get_script().resource_path.get_file()
		print(formatted_message(context, msg, "E"))
		push_error(formatted_message(context, msg, "E"))
		if ESCProjectSettingsManager.get_setting(
			ESCProjectSettingsManager.TERMINATE_ON_ERRORS
		):
			assert(false)
			escoria.get_tree().quit()
```

Reply:
Fixed already, thanks!